main/src/cgeo/geocaching/maps/DirectionDrawer.java

```diff
@@ -0,0 +1,48 @@
+package cgeo.geocaching.maps;
+
+import cgeo.geocaching.location.Geopoint;
+import cgeo.geocaching.maps.interfaces.MapItemFactory;
+import cgeo.geocaching.maps.interfaces.MapProjectionImpl;
+import cgeo.geocaching.settings.Settings;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Point;
+import android.location.Location;
+
+public class DirectionDrawer {
+    private Geopoint currentCoords;
+    private final Geopoint destinationCoords;
+    private final MapItemFactory mapItemFactory;
+
+    private Paint line = null;
+
+    public DirectionDrawer(final Geopoint coords) {
+        this.destinationCoords = coords;
+        this.mapItemFactory = Settings.getMapProvider().getMapItemFactory();
```

Comment:
I revisited this topic today while preparing the merge and changed my opinion. While strictly speaking problematic, the same pattern is applied in a number of similar places. Therefore I decided it is likely not worth the added overhead of wiring that through several levels in the responsibility chain at this time.